compiler/include/llvmUtil.h

```diff
@@ -26,7 +26,13 @@
 
 #include "llvm/Config/llvm-config.h"
 
-#if   LLVM_VERSION_MAJOR>3 || (LLVM_VERSION_MAJOR==3 && LLVM_VERSION_MINOR>=7 )
+#if   LLVM_VERSION_MAJOR>4 || (LLVM_VERSION_MAJOR==4 && LLVM_VERSION_MINOR>=0 )
+#define HAVE_LLVM_VER 40
+#elif LLVM_VERSION_MAJOR>3 || (LLVM_VERSION_MAJOR==3 && LLVM_VERSION_MINOR>=9 )
+#define HAVE_LLVM_VER 39
+#elif LLVM_VERSION_MAJOR>3 || (LLVM_VERSION_MAJOR==3 && LLVM_VERSION_MINOR>=8 )
+#define HAVE_LLVM_VER 38
+#elif LLVM_VERSION_MAJOR>3 || (LLVM_VERSION_MAJOR==3 && LLVM_VERSION_MINOR>=7 )
 #define HAVE_LLVM_VER 37
 #elif LLVM_VERSION_MAJOR>3 || (LLVM_VERSION_MAJOR==3 && LLVM_VERSION_MINOR>=6 )
 #define HAVE_LLVM_VER 36
```

Comment:
I don't think it's likely that LLVM will use a .10 version any time soon, since they just switched to whole-number-increments every release.

At the same time though, we could make the macro better. In the future I hope to combat its growth by only supporting LLVM 4.0 (say).

Nonetheless I'll add a comment with your suggestion.